### PR TITLE
teach testing & refactoring

### PR DESCRIPTION
**Pull request by katrinleinweber**
- [x] resolve merge conflict offline
- [x] test on Windows

### CODE REVIEW

File: _episodes_rmd/04-making-packages-R.Rmd [hunk 1]
```diff
@@ -6,14 +6,17 @@ questions:
 - "How do I collect my code together so I can reuse it and share it?"
 - "How do I make my own package(s)?"
 - "How do I integrate my function documentation into R's help pages system?"
+- "What helper tools does RStudio provide me with?"
 objectives:
 - "Describe the required structure of R packages."
 - "Create the required structure of a simple R package."
-- "Using roxygen2 to convert function docu into help pages."
+- "Build and install the package in RStudio."
+- "Use roxygen2 to convert function docu into help pages."
```
Comment (lwjohnst86) [inline _episodes_rmd/04-making-packages-R.Rmd]: docu -> document

Comment (lwjohnst86) [reply _episodes_rmd/04-making-packages-R.Rmd]: Unless there is a reason it says "docu"... not sure why though. It seems that this docu word is used throughout.

File: _episodes_rmd/05-testthat.Rmd [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+more readable for collaborators, reviewers, etc. You probably know the saying
+"Never change a running system!" We surely invested a lot of time already to make
+sure our functions work fine now. It is natural to be cautious about changing
+computer code and it rightly shouldn't be done on a whim.
+
+However, as you have probably learned in the Git lesson, branching is one way to
+try out code changes in a way that allows you to recover from mistake, and only
+merge successful changes.
+
+Unit test are another way to "span a safety net", and in R, the [`testthat` package][testthat]
```
Comment (lwjohnst86) [inline _episodes_rmd/05-testthat.Rmd]: "Unit testing is" or "Unit tests are"

File: _episodes_rmd/05-testthat.Rmd [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+
+Remember that we updated `rescale()` with lower and upper bounds and default
+values at the end of [episode 2][02]? We had to manually test that change with a
+new example back then. We were repeating ourselves a bit more often than necessary
+back then, weren't we? Of course, there are ways to automate the testing of code
+changes, and to give you quick feedback whether your changes worked, or broke
+anything.
+
+To give ourselves this quick feedback, use `testthat`'s `auto_test_package()`
+and notice the hopefully all green and `OK` `Results`.
```
Comment (lwjohnst86) [inline _episodes_rmd/05-testthat.Rmd]: Or use Ctrl-Shift-T in RStudio for running tests. Or do you mention that already? I haven't seen it yet.

File: _episodes_rmd/05-testthat.Rmd [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+bias when coding. If one already worked hard to get the code to run, one may be
+_biased to_ let the tests _confirm_ what the code currently _does_, instead of
+challenging it with tests about what it _should_ do. The latter is similar to the
+scientific method of trying to disprove hypotheses, so that the truth remains.
+
+[02]: {{ page.root }}/03-func-R/
+[TDD]: https://en.wikipedia.org/wiki/Test-driven_development
+
+> ## Update `test-rescale.R` with the second example from the roxygen comments at the end of
+[episode 2][02].
```
Comment (lwjohnst86) [inline _episodes_rmd/05-testthat.Rmd]: Will this render properly if on a new line? I don't think it will.

**Review by lwjohnst86 — CHANGES_REQUESTED** [review]
Looking good! Minor edits/comments

File: _episodes_rmd/05-testthat.Rmd (hunk 4 above)
Comment (katrinleinweber) [reply _episodes_rmd/05-testthat.Rmd]: Fix on another branch in a larger link-update commit.

**Comment by katrinleinweber** [comment]
Thanks for the corrections, @lwjohnst86 :-)

**Comment by lwjohnst86** [comment]
;)

**Pull request merged**